active-directory-new-domain-sql/azuredeploy.json

```diff
@@ -0,0 +1,708 @@
+    "adSubnet": {
+      "type": "string",
+      "metadata": {
+        "description": "The address range of the subnet created in the new VNET"
+      },
+      "defaultValue": "10.0.0.0/24"
+    },
+    "adPDCNicName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the NIC attached to the new PDC"
+      },
+      "defaultValue": "adPDCNic"
+    },
+    "adPDCNicIPAddress": {
+      "type": "string",
+      "metadata": {
+        "description": "The IP address of the new AD PDC"
+      },
+      "defaultValue": "10.0.0.4"
+    },
+    "adSQLNicName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the NIC attached to the new SQL"
+      },
+      "defaultValue": "adSQLNic"
+    },
+    "adSQLNicIPAddress": {
+      "type": "string",
+      "metadata": {
+        "description": "The IP address of the new AD SQL"
+      },
+      "defaultValue": "10.0.0.5"
+    },
+    "publicIPAddressName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the public IP address used by the Load Balancer"
+      },
+      "defaultValue": "adpublicIP"
+    },
+    "publicIPAddressType": {
+      "type": "string",
+      "allowedValues": [
+        "Dynamic",
+        "Static"
+      ],
+      "metadata": {
+        "description": "The type of the public IP address used by the Load Balancer"
+      },
+      "defaultValue": "Dynamic"
+    },
+    "adPDCVMName": {
+      "type": "string",
+      "metadata": {
+        "description": "The computer name of the PDC"
+      },
+      "defaultValue": "adPDC"
+    },
+    "adSQLVMName": {
+      "type": "string",
+      "metadata": {
+        "description": "The computer name of the SQL"
+      },
+      "defaultValue": "adSQL"
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Administrator of the new VM and Domain"
+      },
+      "defaultValue": "adAdministrator"
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the Administrator account of the new VM and Domain"
+      },
+	"defaultValue": "Pa##w0rd"
+    },
+    "adVMSize": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_D1",
+        "Standard_DS1",
+        "Standard_D2",
+        "Standard_DS2",
+        "Standard_D3",
+        "Standard_DS3",
+        "Standard_D4",
+        "Standard_DS4",
+        "Standard_D11",
+        "Standard_DS11",
+        "Standard_D12",
+        "Standard_DS12",
+        "Standard_D13",
+        "Standard_DS13",
+        "Standard_D14",
+        "Standard_DS14"
+      ],
+      "metadata": {
+        "description": "The size of the VM Created"
+      },
+      "defaultValue": "Standard_D2"
+    },
+    "imagePublisher": {
+      "type": "string",
+      "defaultValue": "MicrosoftWindowsServer",
+      "metadata": {
+        "description": "Image Publisher"
```

Comment:
Image properties can potentially become variables as well as the template scenario is dependent on Windows Server. You could give an option to select a specific SKU (2012 or another) as a parameter below